blog/_posts/2018-04-12-in-a-nutshell.md

```diff
@@ -0,0 +1,175 @@
+the `dotc` compiler which are then pickled and unpickled in the Tasty
+file format. An interesting alternative would be to write Tasty
+picklers and unpicklers that work directly with reflect trees.
+
+Then, we need to define and implement semantic operations such as
+
+ - what are the members that can be selected on this expression?
+ - which subclasses are defined for a sealed trait?
+ - does this expression conform to some expected type?
+
+Finally, we need to connect the new reflection layer to the existing
+high-level macro system. This looks not very difficult. In essence, we
+need to define a pair of mappings between high level trees of type
+`scala.quoted.Expr[T]` and lower-level Tasty trees of type
+`tasty.Term`. Mapping a high-level tree to a low-level one simply
+means exposing its structure. Mapping a a low-level tree to a
+high-level tree of type `scala.quoted.Expr[T]` means checking that the
+low-level tree has indeed the given type `T`. That should be all.
+
+## Future Macros
+
+Adopting this scheme gives already some idea what Scala 3 macros will
+look like. First, they will run after the typechecking phase is
+finished because that is when Tasty trees are generated and
+consumed. This means macros will be blackbox - a macro expansion
+cannot influence the type of the expanded expression as seen from the
+typechecker. As long as that constraint is satisfied we should be able
+to support both `def` macros and annotation macros.
+
+For instance, one could define an annotation macro `@json` that adds a
+JSON serializers to a type. The difference to
+today's "macro paradise" annotation macros (which are not part of the
+official Scala distributon) is that in Scala 3 the generated
+serializers can be seen only in downstream projects, because the
+expansion driven by the annotation happens after type checking.
+
+We might support some forms of whitebox macros by allowing macros in
+the types themselves. These macros would be highlevel only, and would
+integrate with implicit search. A sketch of such as system is outlined
+in [Dotty PR 3844](https://github.com/lampepfl/dotty/pull/3844).
+
+The Scala 3 language should also directly incorporate some constructs
+that so far required advanced macro code to define. For instance:
+
+ - We model lazy implicits directly using
+[by-name parameters](http://dotty.epfl.ch/docs/reference/implicit-by-name-parameters.html) instead of through a complicated macro.
+ - Native [type lambdas](http://dotty.epfl.ch/docs/reference/type-lambdas.html) reduce the need for
+   [kind projector](https://github.com/non/kind-projector).
+ - We also plan to offer the basic building blocks for [generic programming](https://github.com/milessabin/shapeless) and [typeclass derivation](https://github.com/propensive/magnolia).
+
+## Please Give Us Your Feedback!
+
+What do you think of the macro roadmap? Your feedback would be much
+appreciated. There is also lots of scope to shape the future by
```

Comment:
I would suggest linking to a https://contributors.scala-lang.org thread that you create just before this is merged, so it's clear where feedback should go.